thread.go

```diff
@@ -0,0 +1,47 @@
+package notmuch
+
+// Copyright © 2015 The go.bindings Authors. Authors can be found in the AUTHORS file.
+// Licensed under the GPLv3 or later.
+// See COPYING at the root of the repository for details.
+
+// #cgo LDFLAGS: -lnotmuch
+// #include <stdlib.h>
+// #include <notmuch.h>
+import "C"
+
+type (
+	// Thread represents a notmuch thread.
+	Thread struct {
+		id      string
+		cptr    *C.notmuch_thread_t
+		threads *Threads
+	}
+
+	// Messages represents notmuch messages.
+	Messages struct {
+		cptr   *C.notmuch_messages_t
+		thread *Thread
+	}
+)
+
+// GetSubject returns the subject of a thread.
+func (t *Thread) GetSubject() string {
+	cstr := C.notmuch_thread_get_subject(t.toC())
+	str := C.GoString(cstr)
+	return str
+}
+
+// GetID returns the ID of the thread.
+func (t *Thread) GetID() string {
+	return t.id
+}
+
+// Next retrieves the next message from the thread. Next returns true if a message
+// was successfully retrieved.
+func (t *Thread) Next(m *Message) {
```

Comment:
So, obviously this is unimplemented and the signature disagrees with the comment. Also, this doesn't appear to wrap a notmuch function.

I'd say let's just defer adding additional wrappers to another pr, but my $0.02: I'd have a method on `*Thread` that returns a `*Messages` instead.